packages/store/src/-private/record-arrays/identifier-array.ts

```diff
@@ -355,11 +362,9 @@ class IdentifierArray {
               }
               const args: unknown[] = Array.prototype.slice.call(arguments);
               assert(`Cannot start a new array transaction while a previous transaction is underway`, !transaction);
+
               transaction = true;
-              let result: unknown = Reflect.apply(target[prop] as ProxiedMethod, receiver, args);
-              self[MUTATE]!(prop as string, args, result);
-              addToTransaction(_TAG);
-              // TODO handle cache updates
+              const result = self[MUTATE]!(target, receiver, prop as string, args, _TAG);
```

Comment:
one thing I've been constantly debating around this line is whether we should try/finally it to ensure transaction is set back to false if we error.